### ISSUE TITLE
[Bug] When building image with RUN --mount type=tmpfs,dst=/tmp final image doesn't contain /tmp directory

### ISSUE THREAD

**Issue by rahilarious** [containers/podman#18183]
### Issue Description

```
FROM docker.io/library/debian:testing-slim
RUN  --mount=type=tmpfs,dst=/tmp \
     --mount=type=tmpfs,dst=/var/tmp \
    echo '#####        I am sandboxed in container just like you are in matrix      ######'

CMD ["/usr/bin/bash"]
```

### Steps to reproduce the issue

Steps to reproduce the issue
1. Build above Containerfile.
2. Check final image if if contains /tmp directory
3. Check `docker.io/library/debian:testing-slim` image if it contains /tmp dir
4. Bug??

### Describe the results you received

No /tmp dir found in final image, but /var/tmp is there !!!
```
$ podman diff docker.io/library/debian:testing-slim e26d5b515b9f
C /etc
A /tmp
```

### Describe the results you expected

Image should also contain /tmp dir

### podman info output

```yaml
host:                  
  arch: amd64    
  buildahVersion: 1.29.0                                                                                                                                      
  cgroupControllers:
  - cpuset     
  - cpu                          
  - io        
  - memory                
  - hugetlb                                                                                                                                                   
  - pids           
  - misc                                                                       
  cgroupManager: systemd                                                       
  cgroupVersion: v2        
  conmon:               
    package: conmon-2.1.7-2.fc38.x86_64 
    path: /usr/bin/conmon           
    version: 'conmon version 2.1.7, commit: '
  cpuUtilization:
    idlePercent: 98.03         
    systemPercent: 0.99                                                        
    userPercent: 0.98  
  cpus: 4                                                                      
  distribution:        
    distribution: fedora
    version: "38"      
  eventLogger: journald                                                        
  hostname: lenovo
  idMappings:        
    gidmap: null
    uidmap: null     
  kernel: 6.2.9-300.fc38.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 716091392                                                                                                                                          
  memTotal: 3958779904                                                                                                                                        
  networkBackend: netavark                                                                                                                                    
  ociRuntime:                                                                                                                                                 
    name: crun                                                                                                                                                
    package: crun-1.8.3-2.fc38.x86_64                                                                                                                         
    path: /usr/bin/crun                                                        
    version: |-                    
      crun version 1.8.3            
      commit: 59f2beb7efb0d35611d5818fd0311883676f6f7e                         
      rundir: /run/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL                                                                     
  os: linux         
  remoteSocket:
    path: /run/podman/podman.sock
  security:   
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID                
    rootless: false
    seccompEnabled: true                                                       
    seccompProfilePath: /usr/share/containers/seccomp.json                     
    selinuxEnabled: false  
  serviceIsRemote: false
  slirp4netns:                                                                 
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-12.fc38.x86_64
    version: |-  
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383                         
      libslirp: 4.7.0  
      SLIRP_CONFIG_VERSION_MAX: 4                                              
      libseccomp: 2.5.3
  swapFree: 3958366208  
  swapTotal: 3958370304
  uptime: 29h 2m 11.00s (Approximately 1.21 days)                              
plugins:          
  authorization: null
  log:          
  - k8s-file         
  - none                       
  - passthrough    
  - journald         
  network:                                                                                                                                                    
  - bridge                                                                                                                                                    
  - macvlan                                                                                                                                                   
  volume:                                                                                                                                                     
  - local                                                                                                                                                     
registries:                                                                                                                                                   
  search:                                                                      
  - registry.fedoraproject.org     
  - registry.access.redhat.com      
  - docker.io                                                                  
  - quay.io            
store:           
  configFile: /etc/containers/storage.conf                                                                                                                    
  containerStore:   
    number: 1  
    paused: 0                    
    running: 1
    stopped: 0            
  graphDriverName: btrfs                                                                                                                                      
  graphOptions: {} 
  graphRoot: /var/lib/containers/storage                                       
  graphRootAllocated: 429496729600                                             
  graphRootUsed: 3281801216
  graphStatus:          
    Build Version: Btrfs v6.1.3                                                
    Library Version: "102"          
  imageCopyTmpDir: /var/tmp                                                    
  imageStore:    
    number: 9                  
  runRoot: /run/containers/storage                                             
  transientStore: false
  volumePath: /var/lib/containers/storage/volumes                              
version:               
  APIVersion: 4.4.2     
  Built: 1677669759    
  BuiltTime: Wed Mar  1 16:52:39 2023                                          
  GitCommit: ""   
  GoVersion: go1.20.1
  Os: linux     
  OsArch: linux/amd64
  Version: 4.4.2
```

### Upstream Latest Release
No

**Comment by flouthoc** [comment]
Dupe of https://github.com/containers/buildah/issues/4739

**Comment by rahilarious** [comment]
Can you please mention in which version this was fixed? I tried podman 4.5.0-rc2 buildah 1.30.0 and it is still happening!!
